web-frame/src/app/app-routing-error-handler.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * @license
+ * Copyright (c) 2018-2020 THREEANGLE SOFTWARE SOLUTIONS SRL
+ * Available under MIT license webFrame/LICENSE
+ */
+import { HttpErrorResponse } from '@angular/common/http';
+import { NavigationExtras, Router } from '@angular/router';
+import { trimStart } from 'lodash';
+import { ENVIRONMENT } from '~app-shared/config';
+import { AccessDeniedError, HttpStatusCode, PAGE_URL, PageNotFoundError, WebFrameErrorHandlerService } from '~app-shared/core';
+
+export function appRoutingErrorHandler(this: Router, err: unknown): unknown {
+  const extras: NavigationExtras = { skipLocationChange: true };
+  if (err instanceof AccessDeniedError) {
+    return this.navigateByUrl(PAGE_URL.ACCESS_DENIED_PAGE, extras);
+  } else if (err instanceof PageNotFoundError) {
+    return this.navigateByUrl(PAGE_URL.PAGE_NOT_FOUND_PAGE, extras);
+  } else if (err instanceof HttpErrorResponse) {
+    if (err.status === HttpStatusCode.FORBIDDEN) {
+      return this.navigateByUrl(PAGE_URL.ACCESS_DENIED_PAGE, extras);
+    } else if (err.status >= HttpStatusCode.BAD_REQUEST
+      && err.status < HttpStatusCode.INTERNAL_SERVER_ERROR) {
+      return this.navigateByUrl(PAGE_URL.PAGE_NOT_FOUND_PAGE, extras);
+    } else if (err.status >= HttpStatusCode.INTERNAL_SERVER_ERROR) {
+      try {
+        const errorString: string = JSON.stringify(err);
+        window.sessionStorage.setItem(WebFrameErrorHandlerService.LAST_RUNTIME_ERROR_KEY, errorString);
```

Comment:
This is like a magic key that you need to know it's stored from this location.
Why not create a method on WebFrameErrorHandlerService that implements these details behind the scene, so everything is unified under the same service.

```typescript
try {
   WebFrameErrorHandlerService.updateLastRuntimeError(error)
}

```

Reply:
* created 3 helper functions: `isAccessDeniedError`, `isPageNotFoundError` and `isInternalServerError`.
* Replaced the "go to error page" logic with a call to this static method: `WebFrameErrorHandlerService.DISPLAY_STANDALONE_ERROR_PAGE()`

Reply:
Sorry I cannot see how the response is related to what I wrote. Maybe you wanted to reply to a different comment

Reply:
The call to SessionStorage was only needed because the routing error handler was also performing the redirect to the error page.


I have now converted all relevant methods to static and instead of using the `LAST_RUNTIME_ERROR_KEY` static property, the sessions storage call is performed by a `DISPLAY_STANDALONE_ERROR_PAGE()` static method call. Now there is no need to create a `updateLastRuntimeError()` method, it's now a hidden implementation detail.